server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -47,8 +47,6 @@ public class Version implements Comparable<Version>, ToXContentFragment {
      */
     private static final int V_EMPTY_ID = 0;
     public static final Version V_EMPTY = new Version(V_EMPTY_ID, 0, false, org.apache.lucene.util.Version.LATEST);
-    public static final Version V_3_0_1 = new Version(6_01_04_99, 3_00_01_99, false, org.apache.lucene.util.Version.LUCENE_7_1_0);
```

Comment:
Can we keep the entry to keep the versions in the places where it was used?

It's possible to create the lucene version with the `fromBits` static factory if the constant is gone.